File: 1721-swapping-nodes-in-a-linked-list/1721-swapping-nodes-in-a-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* swapNodes(ListNode* head, int k) {
        
        vector<int> A;
        ListNode *temp = head;
        while(temp != NULL)
        {
            A.push_back(temp->val);
            temp = temp->next;
        }
        int a;
        int n = A.size();
        a = A[n-k];
        A[n-k] = A[k-1];
        A[k-1] = a;
        temp = head;
        int  i = 0;
        while(temp != NULL)
        {
            temp->val = A[i];
            i++;
            temp = temp -> next;
            
        }
        
        return head;
        
        
    }
};

File: 1721-swapping-nodes-in-a-linked-list/1721-swapping-nodes-in-a-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode swapNodes(ListNode head, int k) {
        
        List<ListNode> op =new ArrayList<>();
        ListNode ptr = head;
        while(ptr!=null)
        {
            op.add(ptr);
            ptr = ptr.next;
        }
        int n = op.size();
        int temp = op.get(k-1).val;
        op.get(k-1).val = op.get(n-k).val;
        op.get(n-k).val = temp;
        
        
        return head;
        
        
    }
}